### ISSUE TITLE
numpy.average and numpy.mean give different results for masked arrays

### ISSUE THREAD

**Issue by metaperture** [numpy/numpy#8332]
```
a = np.ma.array([10, 20, 30, 40, 50, 60, 70, 80, 90, -1])
log_a = np.log(a)
np.average(log_a), np.mean(log_a) # returns (3.452509331702788, 3.7250103685586535)
```
This can be fixed by changing https://github.com/numpy/numpy/blob/v1.11.0/numpy/lib/function_base.py#L921 from "np.asarray" to "np.asanyarray", but I didn't want to make a whole dev env just for that little fix.

**Comment by mhvk** [comment]
Thanks for the report. This was a known annoyance that should be fixed in recent numpy (and will be available in numpy 1.13); see #8290.

**Comment by metaperture** [comment]
Oh thanks, didn't see that when I searched.
